src/Widgets/AgendaEventRow.vala

```diff
@@ -35,12 +35,126 @@ public class Maya.View.AgendaEventRow : Gtk.ListBoxRow {
+                    ///The number of words can differ from US English and need not be a direct translation.
+                    return _("car;drive;driving;road trip;");
+
+                case FLIGHT:
+                    ///Translators: Give a list of air travel related keywords, separated by semicolons.
+                    ///The number of words can differ from US English and need not be a direct translation.
+                    return _("flight;airport;");
+
+                case FOOD:
+                    ///Translators: Give a list of food consumption related keywords, separated by semicolons.
+                    ///The number of words can differ from US English and need not be a direct translation.
+                    return _("breakfast;brunch;dinner;lunch;supper;steakhouse;burger;meal;barbecue");
+
+                case LEGAL:
+                    ///Translators: Give a list of law related keywords, separated by semicolons.
+                    ///The number of words can differ from US English and need not be a direct translation.
+                   return _("court;jury;tax;attorney;lawyer;contract");
+
+                case MOVIE:
+                    ///Translators: Give a list of movie (film) related keywords, separated by semicolons.
+                    ///The number of words can differ from US English and need not be a direct translation.
+                    return _("movie");
+
+                case WEDDING:
+                    ///Translators: Give a list of wedding related keywords, separated by semicolons.
+                    ///The number of words can differ from US English and need not be a direct translation.
+                    return _("wedding");
+
+                default:
+                    return "";
+            }
+        }
+
+        public unowned string? get_icon_name () {
+            switch (this) {
+                case APPOINTMENT:
+                    return "event-appointment-symbolic";
+
+                case CELEBRATION:
+                    return "event-birthday-symbolic";
+
+                case CALL:
+                    return "event-call-symbolic";
+
+                case DRINKS:
+                    return "event-cocktails-symbolic";
+
+                case DRIVING:
+                    return "event-driving-symbolic";
+
+                case FLIGHT:
+                    return "event-flight-symbolic";
+
+                case FOOD:
+                    return "event-food-symbolic";
+
+                case LEGAL:
+                    return "event-legal-symbolic";
+
+                case MOVIE:
+                    return "event-movie-symbolic";
+
+                case WEDDING:
+                    return "event-wedding-symbolic";
+
+                default:
+                    return null;
+            }
+        }
+    }
+
+//    private Gee.HashMap<Category, string> category_icon_map;
+    private Gee.HashMultiMap<Category, string> keyword_map; /* Would TreeMultiMap be better? */
```

Comment:
You probably don't want to leave this question remaining forever :)
```suggestion
    private Gee.HashMultiMap<Category, string> keyword_map;
```
As there is no redundancy in key values, a HashMultiMap seems fine

Reply:
Removed comment